test/automation/projects_automation_test.go

```diff
@@ -0,0 +1,87 @@
+package automation
+
+import (
+	"fmt"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/stretchr/testify/suite"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+type ProjectsAutomationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *ProjectsAutomationTestSuite) TestProjects_NoActProjects() {
+	suite.OnlyRunForTags(tagsuite.Automation)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.Spawn("projects")
+	cp.ExpectLongString("You have not activated any projects yet")
+	cp.Wait(5 * time.Second)
+	fmt.Println(cp.Snapshot())
+}
+
+func (suite *ProjectsAutomationTestSuite) TestProjects_LocalChkout() {
+	suite.OnlyRunForTags(tagsuite.Automation)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	// Test with PUBLIC project
+	url := "https://platform.activestate.com/qamainorg/public?branch=main&commitID=32e543ee-b6ab-4f59-9b28-ad830ec6980e"
+	suite.Require().NoError(fileutils.WriteFile(filepath.Join(ts.Dirs.Work, "activestate.yaml"), []byte("project: "+url)))
+
+	cp := ts.Spawn("projects")
+	cp.Expect("Organization")
+	cp.Expect("public")
+	cp.Expect("Local Checkout")
+	cp.ExpectExitCode(0)
+
+	// Test with PRIVATE project
+	url = "https://platform.activestate.com/qamainorg/private?branch=main&commitID=92935f87-cc8f-4da3-82d5-13e3e5249452"
+	suite.Require().NoError(fileutils.WriteFile(filepath.Join(ts.Dirs.Work, "activestate.yaml"), []byte("project: "+url)))
+
+	cp = ts.Spawn("projects")
+	cp.Expect("Organization")
+	cp.Expect("private")
+	cp.Expect("Local Checkout")
+	cp.ExpectExitCode(0)
```

Comment:
Replacing the `activestate.yaml` shouldn't update the project listing... can we run these tests on our CI to ensure that this is passing?

Reply:
Local execution is passed.

Reply:
Labelled by Test:automation